Card update

Duncid opened the pull request:
## Description

Bringing consistency to different "Card" components:
- `Card` > `ValueCard`
- `CardButton` > `Card`
- `ValueCard` makes use of `Card`
- `AssistantPreview` > To be replaced by `AssistantCard`
- `Citation` don't change (already based on `Card`

Creating a new CardGrid component to simply create a responsive Card grid.

## Risk
Needs to be careful at Deploy as components will change names.

## Deploy Plan
Merge, deploy a new Sparkle, new PR for making use of the new Sparkle.

Duncid committed:
WIP

Duncid committed:
refactoring Card and renaming to ValueCard

Duncid committed:
New AssistantCard

Duncid committed:
CardGrid

Duncid committed:
Fixing

JulesBelveze commented on sparkle/src/components/AssistantCard.tsx:
```diff
@@ -0,0 +1,157 @@
+      variant="outline"
+      icon={MoreIcon}
+      className={className}
+      {...props}
```
I don't think you wanna spread the props here

JulesBelveze reviewed (commented):
Left a few comments. I'm not sure if the `ValueCard` component still adds a lot of value, could we derive it from the `Card` itself?

Duncid committed:
Update sparkle/tailwind.config.js

Co-authored-by: Jules Belveze <32683010+JulesBelveze@users.noreply.github.com>

Duncid committed:
PR review fixes

Duncid commented:
Thanks @JulesBelveze !
Made changes accordingly.

Duncid committed:
Iterating

JulesBelveze committed:
[sparkle] - refactor: streamline CardActionButton and import clean-ups

 - Simplified the CardActionButton in the Card component to use fewer default props
 - Moved LinkWrapperProps import to its own line for consistency
 - Corrected the order of responsive grid class names in CitationNew component
 - Updated class order for consistency in SheetHeader component
 - Enhanced ActionCardDemo with a mini sized Command Line icon button

[sparkle] - refactor: adjust class order for better readability

 - Standardized the order of utility classes to improve code consistency and readability across several components

JulesBelveze committed:
[sparkle] - refactor: update Card component props and imports

 - Replace ButtonProps with MiniButtonProps for CardActionButton
 - Consolidate import lines by removing duplicated LinkWrapperProps import
 - Ensure Button imports are correctly adjusted for component dependencies

JulesBelveze committed:
[sparkle] - refactor: update AssistantCardMore to use MiniButtonProps

 - Replaced ButtonProps with Omit<MiniButtonProps, "icon"> to refine prop types in AssistantCard
 - Ensured AssistantCardMore does not include an "icon" prop to adhere to new prop structure

JulesBelveze committed:
[sparkle] - refactor: standardize AssistantCardMore button size

 - Removed 'size' prop from AssistantCardMore type and hard-coded it to "mini" for consistency

JulesBelveze committed:
rebase main

Duncid committed:
Updating

Duncid committed:
Fixing merge

Duncid committed:
fix

Duncid committed:
Removeing AssistantPreview export

Duncid committed:
CitationGrid use in Message

Duncid committed:
did a npx prettier -w . && npx eslint . --fix

Duncid committed:
fix

Duncid committed:
remove

Duncid committed:
Fixing hover

Duncid committed:
Fixing message bottom space

Duncid committed:
Fixing Large assistant card

Duncid committed:
Adding Variant prop to AssistantCard

Duncid committed:
PaginatedCitation update

Duncid committed:
last updates

Duncid committed:
patch

Duncid committed:
patch

JulesBelveze reviewed (approved):
Looks 🔥 thx 🙏🏼
